llvm/lib/CodeGen/ScheduleDAGInstrs.cpp

```diff
@@ -596,6 +596,10 @@ std::optional<unsigned> ScheduleDAGInstrs::initSUnit(MachineInstr &MI) {
   if (MI.isDebugOrPseudoInstr())
     return {};
 
+  // Ensure all instructions can be given a SUnit without re-allocation.
+  if (SUnits.empty())
+    SUnits.reserve(NumRegionInstrs);
```

Comment:
Hmm. This proliferates NumRegionInstrs

Reply:
What do you mean? I'm simply moving the call from `initSUnits` to `initSUnit` so the latter is easier to use. I could also add a `reserverSUnits()` hook, but that seems a bit overkill.

Reply:
It was possible to add SUnits without any context, provided that you didn't use the MI to SU map. AFAIK, that was originally introduced to create the lcd edge DAG in loop aware, where the MI-> SU map can not be used because we have a one-to-many relation there. That may have NumRegionInstrs visible, but non sensibly initialized.

Reply:
Discussed offline, I will reserve capacity in `enterRegion`, see https://github.com/Xilinx/llvm-aie/pull/236#discussion_r1846399685